localstack-core/localstack/aws/handlers/validation.py

```diff
@@ -102,11 +102,14 @@ def __call__(self, chain: HandlerChain, context: RequestContext, response: Respo
             for openapi in self.open_apis:
                 try:
                     openapi.validate_request(WerkzeugOpenAPIRequest(context.request))
+                    # We stop the handler at the first succeeded validation, as the other spec might not even specify
+                    #   this path.
+                    break
                 except RequestValidationError as e:
                     # Note: in this handler we only check validation errors, e.g., wrong body, missing required.
                     response.status_code = 400
                     response.set_json({"error": "Bad Request", "message": str(e)})
-                    chain.stop()
+                    chain.terminate()
```

Comment:
q: You switched to terminate here in order to avoid the execution of the response handlers here. We only ever used this for the CORS handling to explicitly avoid any execution after the CORS handler. Why is this necessary here? This would circumvent f.e. the response logging.

Reply:
No need actually. Thanks for pointing this out.